src/glang/codegen/builtin_types.py

```diff
@@ -710,9 +735,10 @@ def format_for_output_to_user(self) -> str:
     @property
     def ir_mangle(self) -> str:
         assert not self.is_illegal_value_type
-        dimensions = "_".join(map(str, self.known_dimensions))
-        const = "" if self.is_mut else "C"
-        return f"__UA{const}_{self.member.ir_mangle}__{dimensions}__"
+        #  <array-type> ::= A [<array bound number>] _ <element type>
+        known_dimensions = str.join("", map(lambda d: f"A{d}_", self.known_dimensions))
+        const = "" if self.is_mut else "K"
+        return const + "PA_" + known_dimensions + self.member.ir_mangle
```

Comment:
Why is this `PA_` not just `A_`?
https://itanium-cxx-abi.github.io/cxx-abi/abi.html#mangle.array-type

Reply:
I'm guessing c++ just encodes it as a pointer + fixed sized arrays?

Reply:
Yeah, heap arrays decay into pointers (carried over from C no doubt)

Reply:
I think this would be fun to test. I'm sure that mangling into A_i in Graphene and calling from c++ would be fine due to ODR, I'm not convinced we'd be smart enough to link to the correct c++ function from Graphene. Maybe this is all handled in the linker and it doesn't matter.

If it makes no practical difference I'd much prefer A_i personally.

Reply:
Hmm, not sure what you mean - the linker just sees symbols and matches them verbatim. If we want to link with C++ code then we need to produce exactly matching symbols.

> If it makes no practical difference I'd much prefer A_i personally.

Happy to use `A_i` until we have a good reason to do anything else